core/src/edu/asu/c3simulator/simulation/business/Company.java

```diff
@@ -0,0 +1,374 @@
+	 * <p>
+	 * This method is intended to be called by Simulation and its agents, as part of a
+	 * market/economy simulation.
+	 * 
+	 * @param product
+	 *            The product to be bought/sold
+	 * @param quantity
+	 *            The quantity to buy/sell
+	 * @return True if the sale was successful, false if the sale was unable to be
+	 *         processed. Note, a false return is indicitive of the quantity being too
+	 *         high for this company to support a sale. For instance, the quantity may
+	 *         cause the company to exceed its monthly sales potential, as specified by
+	 *         {@link #getMonthlySalePotential(Product)}
+	 */
+	public boolean registerSale(Product product, int quantity)
+	{
+		int salesInCurrentMonth = getThisMonthsSalesOf(product) + quantity;
+		
+		if (salesInCurrentMonth > getMonthlySalePotential(product))
+		{
+			return false;
+		}
+		else
+		{
+			float grossProfit = getRetailPriceOf(product) * quantity;
+			credit.processGrossProfit(grossProfit);
+			this.salesInCurrentMonth.put(product, salesInCurrentMonth);
+			return true;
+		}
+	}
+	
+	/**
+	 * If {@link #salesInCurrentMonth} does not contain the specified product, the product
+	 * will be added to the map and initialized with a value of 0, and that value will be
+	 * returned.
+	 * <p>
+	 * Otherwise, the corresponding value in {@link #salesInCurrentMonth} will be
+	 * returned.
+	 */
+	private int getThisMonthsSalesOf(Product product)
+	{
+		if (salesInCurrentMonth.containsKey(product))
+		{
+			return salesInCurrentMonth.get(product);
+		}
+		else
+		{
+			salesInCurrentMonth.put(product, 0);
+			return 0;
+		}
+	}
+	
+	/**
+	 * Equivalent to {@link #retailPrice}, but handles the case in which
+	 * {@link #retailPrice} is not registered for the given product.
+	 * <p>
+	 * As of version 0.1, all products must register a retail price, so an
+	 * {@link IllegalStateException} is thrown if {@link #retailPrice} does not contain
+	 * the given product. However, this functionality may be changed in future revisions
+	 * and is subject to change.
+	 */
+	private float getRetailPriceOf(Product product)
+	{
+		if (retailPrice.containsKey(product))
+		{
+			return retailPrice.get(product);
+		}
+		else
+		{
+			throw new IllegalStateException(
+					"Retail price must be registered for all products");
+		}
+	}
+	
+	/**
+	 * Register all expenses, process production (as specified by
+	 * {@link #processProduction()}), and update {@link #publicOpinionOfProducts} and
+	 * {@link #publicOpinionOfCompany} as well as all campaigns in
+	 * {@link #activeMarketing}
+	 */
+	public void processMonth()
+	{
+		processProduction();
+		float expense = getMonthlyEmployeeExpense() + getMonthlyMarketingExpense();
+		credit.processPayment(expense);
+		
+		List<MarketingCampaign> completedCampaigns = new LinkedList<>();
+		for (MarketingCampaign campaign : activeMarketing)
+		{
+			Product product = campaign.getTarget();
+			int publicityGain = campaign.processMonth();
```

Comment:
Is this relative to the current publicity or is it absolute? 
Does publicityGain means that publicOpinionOfCompany could only increase or does it just mean change in publicity from the previous for the current month?